inc/msp/FlightController.hpp

```diff
@@ -3,205 +3,349 @@
-
-    bool isFirmwareMultiWii() { return isFirmware(FirmwareType::MULTIWII); }
-
-    bool isFirmwareCleanflight() { return isFirmware(FirmwareType::CLEANFLIGHT); }
-
+    bool connect(const std::string &device, const size_t baudrate=115200, const double& timeout = 0.0);
+    
+    /**
+     * @brief Stops MSP control if active, and disconnects the internal Client object
+     * @return True on success
+     */
+    bool disconnect();
+    
+    /**
+     * @brief Tests connection to a flight controller
+     * @return True if connected
+     */
+    bool isConnected();
+    
+    /**
+     * @brief Sets internal flag to arm the flight controller
+     */
+    void arm();
+    
+    /**
+     * @brief Sets internal flag to disarm the flight controller
+     */
+    void disarm();
+    
     /**
-     * @brief subscribe register callback function that is called when type is received
-     * @param callback pointer to callback function (class method)
-     * @param context class with callback method
-     * @param tp period of timer that will send subscribed requests (in seconds), by default this is 0 and requests are not sent periodically
-     * @return pointer to subscription that is added to internal list
+     * @brief Queries internal flag governing flight controller arming
+     * @return True if arm flag is set
+     */    
+    bool armSet();
+    
+    /**
+     * @brief Sets data structure with all flags governing flight mode
+     * @param mode FlightMode object
+     */
+    void setFlightMode(FlightMode mode);
+    
+    /**
+     * @brief Queries data structure with all flags governing flight mode
+     * @return FlightMode object matchhing current flight mode flags
+     */
+    FlightMode getFlightMode();
+    
+    /**
+     * @brief Sets which instruction source the flight controller should 
+     * listen to. Also starts periodic MSP control message if MSP is selected
+     * @param mode FlightMode object
      */
-    template<typename T, typename C>
-    msp::client::SubscriptionBase* subscribe(void (C::*callback)(const T&), C *context, const double tp = 0.0) {
-        return client.subscribe(callback, context, tp);
+    void setControlSource(ControlSource source);
+    
+    /**
+     * @brief Queries the currently active control source
+     * @return ControlSource object matching current control source
+     */
+    ControlSource getControlSource();
+    
+    /**
+     * @brief Sets roll, pitch, yaw, and throttle values. They are sent immediately
+     * and stored for resending if automatic MSP control is enabled.
+     * @param rpyt An array of doubles representing roll, pitch, yaw, and throttle in that 
+     * order. Values are expected to be in the range -1.0 to +1.0, mapping to 1000us to 2000us
+     * pulse widths.
+     */
+    void setRPYT(std::array<double,4>& rpyt);
+    
+    /**
+     * @brief Method used to generate the Rc message sent to the flight controller 
+     */
+    void generateMSP();
```

Comment:
`setRPYT` is not `const`able because the input is swapped with the internal data. I did make it a universal reference though. 

`generateMSP` actually assembles a RC message based on the RPYT values loaded in with `setRPYT` and the mode parameters. It does have some overlap with the other setRC methods.